pkg/scheduler/framework/plugins/defaultassigner/dynamic_test.go

```diff
@@ -280,11 +280,11 @@ func TestDynamicAssigner_Assign(t *testing.T) {
 				},
 			},
 			want: framework.TargetClusters{
-				BindingClusters: []string{"c1", "c2", "c3", "c4"},
+				BindingClusters: []string{"c1", "c2", "c3"},
 				Replicas: map[string][]int32{
-					utils.GetFeedKey(newDeploymentFeed("f1")): {1, 3, 0, 0}, // already scheduled: 4
+					utils.GetFeedKey(newDeploymentFeed("f1")): {1, 3, 0}, // already scheduled: 4
```

Comment:
IIRC, we're leaving extra 0 replica here intentionally. This is very useful for future horizontally scaling, since we've kept all the needed files, configruations, and even volumes out there.

Correct me if I am wrong.  cc @Garrybest

Reply:
Interesting, I guess you mean some dependency like ConfigMap, Secret or PVC? 

From this point of view, it's useful to propagate 0-replica workload with its dependencies. Some PVC may be bound right before scaling. So it's right, I didn't think about this before.

Reply:
> Interesting, I guess you mean some dependency like ConfigMap, Secret or PVC?
>
> From this point of view, it's useful to propagate 0-replica workload with its dependencies. Some PVC may be bound right before scaling.

Yep. This is also very important for future HPA capabilities. With 0 replica, it will be easy and fast to spin up required replicas.

Reply:
I don't think so. 

A cluster with all replica numbers set to zero is a ambiguous definition. When do we need such a cluster? And why this cluster instead of another?

 If a new cluster appears in the scheduling result, the dependencies required for the replicas should be sent together with the replicas. PVCs and other things can be initialized during the pod pending process, which is expected.

Reply:
If a cluster's replica count is reduced to zero through scaling, there should be a parameter that controls the time to retain its related resources, instead of the default retention of the cluster and objects. Otherwise, many invalid objects will be left behind after a long period of time.

Reply:
I think what @dixudx means is like this:

- Scheduler diapatched Deployment and HPA to member cluster with an 0 replica.
- HPA works in member cluster.
- If we don't dispatch this Deployment and HPA, HPA does not work and could not scale replicas.

Reply:
How about passing a parameter to preserve such clusters in another issue ? such as time duration or maxFreeClusters, as a default behavior would be more confusing

Reply:
> many invalid objects will be left behind after a long period of time.

Yeah, this makes sense.

> How about passing a parameter to preserve such clusters in another issue ? such as time duration or maxFreeClusters, as a default behavior would be more confusing

A feature gate or a param sounds good to me. Maybe we could move this feature in another PR.

And I think this will be solely set on `clusternet-scheduler` side, even if we introduce time duration (or TTL) for 0-replica workloads. `clusternet-scheduler` is responsible to update or clean up those expired 0-replicas.

cc @Garrybest

Reply:
> A feature gate or a param sounds good to me. Maybe we could move this feature in another PR.
> And I think this will be solely set on clusternet-scheduler side, even if we introduce time duration (or TTL) for 0-replica workloads. clusternet-scheduler is responsible to update or clean up those expired 0-replicas.

+1